DictateDemo build fails on main because `MLXCommon` is missing `MLXFast` dependency

## Summary

Following the documented DictateDemo instructions on `main` currently fails at compile time with:

```
/Users/.../Sources/MLXCommon/SDPA.swift:3:8: error: no such module 'MLXFast'
import MLXFast
       ^
```

This prevents `Examples/DictateDemo` from building for users who follow the README as written.

## Repro

From a fresh checkout of `main`:

1. `cd Examples/DictateDemo`
2. `swift build`

The current README says the example should then run via:

3. `.build/debug/DictateDemo`

## Actual result

`swift build` fails while compiling `MLXCommon` because `SDPA.swift` imports `MLXFast`, but SwiftPM does not wire that product into the `MLXCommon` target.

## Expected result

`swift build` should succeed from `Examples/DictateDemo`, and the demo executable should be produced in `.build/debug/DictateDemo`.

## Root cause

`Sources/MLXCommon/SDPA.swift` contains an unconditional:

```swift
import MLXFast
```

and uses `MLXFast.scaledDotProductAttention` / `MLXFast.ScaledDotProductAttentionMaskMode`.

However, in the root `Package.swift`, the `MLXCommon` target currently declares:

```swift
.target(
    name: "MLXCommon",
    dependencies: [
        "AudioCommon",
        .product(name: "MLX", package: "mlx-swift"),
        .product(name: "MLXNN", package: "mlx-swift"),
    ]
)
```

and is missing:

```swift
.product(name: "MLXFast", package: "mlx-swift")
```

`MLXFast` still exists in the pinned upstream `mlx-swift`, so this appears to be a package manifest wiring issue in this repo rather than a user environment problem.

## Verified fix

Adding the missing dependency to `MLXCommon` locally resolves the problem:

```swift
.target(
    name: "MLXCommon",
    dependencies: [
        "AudioCommon",
        .product(name: "MLX", package: "mlx-swift"),
        .product(name: "MLXNN", package: "mlx-swift"),
        .product(name: "MLXFast", package: "mlx-swift"),
    ]
)
```

After that local change:

- `cd Examples/DictateDemo && swift build` succeeds
- the demo executable is produced
- `swift test --skip E2E` still passes in the repo root

## Notes

The Swift 6 `Sendable` warnings in `Sources/AudioCommon/ModelLoader.swift` show up during the build, but they are unrelated to this compile failure.

## Environment

- macOS
- Apple Silicon
- Reproduced on `main` on 2026-04-12

--- a/Package.swift
+++ b/Package.swift
@@ -97,6 +97,7 @@ let package = Package(
                 "AudioCommon",
                 .product(name: "MLX", package: "mlx-swift"),
                 .product(name: "MLXNN", package: "mlx-swift"),
+                .product(name: "MLXFast", package: "mlx-swift"),
             ]
         ),
         .target(

